src/aequitas/flow/methods/preprocessing/label_flipping.py

```diff
@@ -251,8 +264,8 @@ def _label_flipping(self, y: pd.Series, s: Optional[pd.Series], scores: pd.Serie
                 if abs(scores.loc[i]) < self.score_threshold:
                     break
 
-                if (disparity[s.loc[i]] > self.disparity_target and y.loc[i] == 1) or (
-                    disparity[s.loc[i]] < self.disparity_target and y.loc[i] == 0
+                if (disparity[s.loc[i]] > self.disparity_target and y.loc[i] == 0) or (
+                    disparity[s.loc[i]] == 0 and y.loc[i] == 1
                 ):
                     y_flipped.loc[i] = 1 - y.loc[i]
                     disparity = self._calculate_prevalence_disparity(y_flipped, s)
```

Comment:
I don't like that we calculate the disparities again and again... What if we just store the number of instances that we need to flip in the disparity method? This could be calculated by using the `disparity_target` also!